service.go

```diff
@@ -54,87 +54,91 @@ func (svc *Service) GetUser(c echo.Context) (user *User, err error) {
+						"status":       status,
+						"replyEventId": resp.ID,
+					}).Error(result.Error)
+					return
+				}
+				nostrEvent.ReplyId = resp.ID
 
-					if status == nostr.PublishStatusSucceeded {
-						nostrEvent.State = NOSTR_EVENT_STATE_PUBLISH_CONFIRMED
-						nostrEvent.RepliedAt = time.Now()
-						svc.db.Save(&nostrEvent)
-						svc.Logger.WithFields(logrus.Fields{
-							"nostrEventId": nostrEvent.ID,
-							"eventId":      event.ID,
-							"status":       status,
-							"replyEventId": resp.ID,
-							"appId":        nostrEvent.AppId,
-						}).Info("Published reply")
-					} else if status == nostr.PublishStatusFailed {
-						nostrEvent.State = NOSTR_EVENT_STATE_PUBLISH_FAILED
-						svc.db.Save(&nostrEvent)
-						svc.Logger.WithFields(logrus.Fields{
-							"nostrEventId": nostrEvent.ID,
-							"eventId":      event.ID,
-							"status":       status,
-							"replyEventId": resp.ID,
-							"appId":        nostrEvent.AppId,
-						}).Info("Failed to publish reply")
-					} else {
-						nostrEvent.State = NOSTR_EVENT_STATE_PUBLISH_UNCONFIRMED
-						svc.db.Save(&nostrEvent)
-						svc.Logger.WithFields(logrus.Fields{
-							"nostrEventId": nostrEvent.ID,
-							"eventId":      event.ID,
-							"status":       status,
-							"replyEventId": resp.ID,
-							"appId":        nostrEvent.AppId,
-						}).Info("Reply sent but no response from relay (timeout)")
-					}
+				if status == nostr.PublishStatusSucceeded {
+					nostrEvent.State = NOSTR_EVENT_STATE_PUBLISH_CONFIRMED
+					nostrEvent.RepliedAt = time.Now()
+					svc.db.Save(&nostrEvent)
+					svc.Logger.WithFields(logrus.Fields{
+						"nostrEventId": nostrEvent.ID,
+						"eventId":      event.ID,
+						"status":       status,
+						"replyEventId": resp.ID,
+						"appId":        nostrEvent.AppId,
+					}).Info("Published reply")
+				} else if status == nostr.PublishStatusFailed {
+					nostrEvent.State = NOSTR_EVENT_STATE_PUBLISH_FAILED
+					svc.db.Save(&nostrEvent)
+					svc.Logger.WithFields(logrus.Fields{
+						"nostrEventId": nostrEvent.ID,
+						"eventId":      event.ID,
+						"status":       status,
+						"replyEventId": resp.ID,
+						"appId":        nostrEvent.AppId,
+					}).Info("Failed to publish reply")
+				} else {
+					nostrEvent.State = NOSTR_EVENT_STATE_PUBLISH_UNCONFIRMED
+					svc.db.Save(&nostrEvent)
+					svc.Logger.WithFields(logrus.Fields{
+						"nostrEventId": nostrEvent.ID,
+						"eventId":      event.ID,
+						"status":       status,
+						"replyEventId": resp.ID,
+						"appId":        nostrEvent.AppId,
+					}).Info("Reply sent but no response from relay (timeout)")
 				}
-			}()
+			}
 		}
+	}()
+
+	<-ctx.Done()
+	if sub.Relay.ConnectionError != nil {
+		svc.Logger.Errorf("Relay error %v", ctx.Err())
+		return sub.Relay.ConnectionError
 	}
+	if ctx.Err() != context.Canceled {
+		svc.Logger.Errorf("Subscription error %v", ctx.Err())
+		return ctx.Err()
+	}
+	svc.Logger.Info("Exiting subscription.")
```

Comment:
if the connection fails we have to wait and retry the connection. is this unaffected?

Reply:
@bumi I tested it against a locally-running relay. There was an issue which I fixed in https://github.com/getAlby/nostr-wallet-connect/pull/207/commits/dd59887a3d8aaf178e4df05c4f68e9716fec7b97.

We currently do not wait but immediately retry - I added a temporary 10 second sleep to test it and it successfully reconnects after I shut down the relay (killing the connection) and restart it.

If the relay cannot be immediately re-connected to, NWC will close. So maybe we should wait (however we haven't been doing that and it seems to work)